Python/wrfrun/unCMIP6.py

```diff
@@ -0,0 +1,622 @@
+            if ((self.lat==None).all() and (self.lon==None).all()):
+                if ((varname=='tos') or (varname=='siconc') or (varname=='sithick')):
+                    raise ValueError("The first field in the table should be a variable different than tos, siconc or sithick.")
+                self.lat = np.array(ds.lat.values)
+                self.lon = np.array(ds.lon.values)
+            else:
+                if ((varname!='tos') and (varname!='sithick') and (varname!='siconc')):
+                    if not((np.allclose(np.array(ds.lat.values),self.lat,rtol=0.0,atol=1.0e-12)) and \
+                        (np.allclose(np.array(ds.lon.values),self.lon,rtol=0.0,atol=1.0e-12))):
+                        raise ValueError("Error: Inconsistent lat/lon values between different fields.")
+            # plev handeling for 3D fields
+            if lvltype=='3d':
+                if ((self.plev==None).all()):
+                    if (varname=='ua') or (varname=='va'):
+                        raise ValueError("The first 3D field in the table should be a variable different than ua and va.")
+                    self.plev = np.array(ds.plev.values)
+                else:
+                    if not(np.allclose(np.array(ds.plev.values),self.plev,rtol=0.0,atol=1.0e-12)):
+                        raise ValueError("Error: Inconsistent pressure levels between 3D fields.")            
+            # Find the date to read
+            if (self.filestrdates[c]!=''):
+                avail_times_t = ds[varname].time.values
+                avail_times = [pd.to_datetime(ele) for ele in avail_times_t]
+                if itm['approx_dates']==False: 
+                    if self.input_date in avail_times:
+                        if avail_times.count(self.input_date)!=1:
+                            raise ValueError('Error: Number of times that date/time appears in file is not 1.')
+                        cdate = self.input_date
+                    else:
+                        raise ValueError('Error: Could not find the exact date and time in the file.')
+                else:
+                    if (not(sorted(avail_times)==avail_times)):
+                        raise ValueError('Error: File times are not sorted.')
+                    if self.input_date<pd.to_datetime(avail_times[0]):
+                        cdate = pd.to_datetime(avail_times[0])    
+                    elif pd.to_datetime(avail_times[-1])<self.input_date:
+                        cdate = pd.to_datetime(avail_times[-1])
+                    else:
+                        if self.input_date in avail_times:
+                            if avail_times.count(self.input_date)!=1:
+                                raise ValueError('Error: Number of times that date/time appears in file is not 1.')
+                            cdate = self.input_date
+                        else:
+                            ndatefound2 = 0
+                            for j in range(len(avail_times)-1):
```

Comment:
if this was actually necessary, I  probably would have used either np.searchsorted (since you already established monotonocity) or used np.argmin the diff between avail_times and input_date - vectorized and much less code.

Reply:
We talked about this. This whole code part is now gone.